<!-- dual_arm_collaboration/config/dual_arm.srdf.xacro -->
<?xml version="1.0" encoding="UTF-8"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="dual_arm_system">
  
  <!-- 規劃群組定義 -->
  <!-- UR5 規劃群組 -->
  <group name="ur5_manipulator">
    <joint name="ur5_shoulder_pan_joint"/>
    <joint name="ur5_shoulder_lift_joint"/>
    <joint name="ur5_elbow_joint"/>
    <joint name="ur5_wrist_1_joint"/>
    <joint name="ur5_wrist_2_joint"/>
    <joint name="ur5_wrist_3_joint"/>
  </group>
  
  <!-- ARM0710 規劃群組 -->
  <group name="arm0710_manipulator">
    <joint name="arm0710_joint1"/>
    <joint name="arm0710_joint2"/>
    <joint name="arm0710_joint3"/>
    <joint name="arm0710_joint4"/>
    <joint name="arm0710_joint5"/>
    <joint name="arm0710_joint6"/>
  </group>
  
  <!-- 雙臂整合群組 -->
  <group name="dual_arms">
    <group name="ur5_manipulator"/>
    <group name="arm0710_manipulator"/>
  </group>
  
  <!-- 末端執行器群組 -->
  <group name="ur5_end_effector">
    <link name="ur5_ee_link"/>
  </group>
  
  <group name="arm0710_end_effector">
    <link name="arm0710_end_effector"/>
  </group>
  
  <!-- 預設姿態定義 -->
  <group_state name="ur5_home" group="ur5_manipulator">
    <joint name="ur5_shoulder_pan_joint" value="0"/>
    <joint name="ur5_shoulder_lift_joint" value="-1.5708"/>
    <joint name="ur5_elbow_joint" value="0"/>
    <joint name="ur5_wrist_1_joint" value="-1.5708"/>
    <joint name="ur5_wrist_2_joint" value="0"/>
    <joint name="ur5_wrist_3_joint" value="0"/>
  </group_state>
  
  <group_state name="arm0710_home" group="arm0710_manipulator">
    <joint name="arm0710_joint1" value="0"/>
    <joint name="arm0710_joint2" value="0"/>
    <joint name="arm0710_joint3" value="0"/>
    <joint name="arm0710_joint4" value="0"/>
    <joint name="arm0710_joint5" value="0"/>
    <joint name="arm0710_joint6" value="0"/>
  </group_state>
  
  <!-- 禁用碰撞檢測的連桿對 -->
  <disable_collisions link1="world" link2="ur5_base_link" reason="Adjacent"/>
  <disable_collisions link1="world" link2="arm0710_base_link" reason="Adjacent"/>
  <disable_collisions link1="ur5_base_link" link2="ur5_shoulder_link" reason="Adjacent"/>
  <disable_collisions link1="arm0710_base_link" link2="arm0710_link1" reason="Adjacent"/>
  
  <!-- 自碰撞檢測禁用 -->
  <disable_collisions link1="ur5_shoulder_link" link2="ur5_upper_arm_link" reason="Adjacent"/>
  <disable_collisions link1="ur5_upper_arm_link" link2="ur5_forearm_link" reason="Adjacent"/>
  <disable_collisions link1="ur5_forearm_link" link2="ur5_wrist_1_link" reason="Adjacent"/>
  <disable_collisions link1="ur5_wrist_1_link" link2="ur5_wrist_2_link" reason="Adjacent"/>
  <disable_collisions link1="ur5_wrist_2_link" link2="ur5_wrist_3_link" reason="Adjacent"/>
  
  <disable_collisions link1="arm0710_link1" link2="arm0710_link2" reason="Adjacent"/>
  <disable_collisions link1="arm0710_link2" link2="arm0710_link3" reason="Adjacent"/>
  <disable_collisions link1="arm0710_link3" link2="arm0710_link4" reason="Adjacent"/>
  <disable_collisions link1="arm0710_link4" link2="arm0710_link5" reason="Adjacent"/>
  <disable_collisions link1="arm0710_link5" link2="arm0710_link6" reason="Adjacent"/>
  
</robot>